web_src/js/components/DiffFileTreeItem.vue

```diff
@@ -1,15 +1,81 @@
 <script lang="ts" setup>
 import {SvgIcon, type SvgName} from '../svg.ts';
 import {diffTreeStore} from '../modules/stores.ts';
-import {ref} from 'vue';
-import type {Item, File, FileStatus} from '../utils/filetree.ts';
+import {computed, nextTick, ref, watch} from 'vue';
+import type {Item, DirItem, File, FileStatus} from '../utils/filetree.ts';
 
-defineProps<{
+// ------------------------------------------------------------
+// Component Props
+// ------------------------------------------------------------
+const props = defineProps<{
   item: Item,
+  setViewed?:(val: boolean) => void,
 }>();
 
+// ------------------------------------------------------------
+// Reactive State & Refs
+// ------------------------------------------------------------
 const store = diffTreeStore();
-const collapsed = ref(false);
+const count = ref(0);
+let pendingUpdate = 0;
+let pendingTimer: Promise<void> | undefined;
+// ------------------------------------------------------------
+// Batch Update Mechanism
+// ------------------------------------------------------------
+/**
+ * Handles batched updates to count value
+ * Prevents multiple updates within the same tick
+ */
+const setCount = (isViewed: boolean) => {
+  pendingUpdate += (isViewed ? 1 : -1);
+
+  if (pendingTimer === undefined) {
+    pendingTimer = nextTick(() => {
+      count.value = Math.max(0, count.value + pendingUpdate);
+      pendingUpdate = 0;
+      pendingTimer = undefined;
+    });
+  }
+};
+
+// ------------------------------------------------------------
+// Computed Properties
+// ------------------------------------------------------------
+/**
+ * Determines viewed state based on item type:
+ * - Files: Directly use IsViewed property
+ * - Directories: Compare children count with viewed count
+ */
+const isViewed = computed(() => {
+  return props.item.isFile ? props.item.file.IsViewed : (props.item as DirItem).children.length === count.value;
```

Comment:
Batching the update sounds like over-engineering to me.
After all, if I read the code correctly, there's a chance of flickering for a frame.
Why not simply
```suggestion
  return props.item.isFile ? props.item.file.IsViewed : (props.item as DirItem).children.every(child => child.file.IsViewed);
```
?

Reply:
The code(`props.item.isFile ? props.item.file.IsViewed : (props.item as DirItem).children.every(child => child.file.IsViewed);`) fails to accurately retrieve folder statuses in deeply nested folder hierarchies, as the isViewed state isn't backend-maintained but dynamically computed on the fly.

This explains why I initially implemented it this way:
```js
export function fileIsViewed(item: Item): boolean {
  return item.isFile ? item.file.IsViewed : (item as DirItem).children.every(fileIsViewed);
}
```

Reply:
Ah, right.
However, that difference should be mainly for historical reasons:
Prior to this PR, the `Viewed` state didn't make sense on directories, so `DirItem` didn't have the attribute.
Now, it does make sense.
So we should be able to move the attribute one layer up into `FileItem` and `DirItem` and convert it from computed on demand into an actual attribute we update.
Then, every tree item has this attribute and we can use this approach.

Reply:
As discussed, moving the folder status calculation logic to the backend would require:

​1. Backend complexity increase: Implementing similar computational logic as in this PR, adding architectural overhead 
2. ​Frontend adaptation: Introducing new logic to synchronize folder read status updates


In contrast, the current PR's approach:

​1. Zero backend changes: Maintains existing backend architecture 
​2. Frontend-driven computation: Handles folder status calculation entirely on the client-side

The biggest difference between the two is that the front - end needs to add a new logic of "reading the folder status from the back - end". The reason why the file status doesn't need to be read separately is that the interaction is initiated by the front - end, so the front - end can directly read its own status. However, if the folder status is calculated by the back - end, a new logic has to be added for reading.

I’m not sure if I’ve made myself clear.